File: Objective-C/Reference App/3. Find Lat Long from Street Address/FindLatLongViewController.m
//
//  FindLatLongViewController.m
//  Reference App
//
//  Created by David Deller on 5/3/18.
//  Copyright © 2018 TallyGo. All rights reserved.
//

#import "FindLatLongViewController.h"

@interface FindLatLongViewController ()

@end

@implementation FindLatLongViewController

- (void)handleInput:(NSString *)input {
    [self findCoordinateFromAddress:input];
}

- (void)findCoordinateFromAddress:(nonnull NSString *)address {
    CLLocationCoordinate2D currentLocation = CLLocationCoordinate2DMake(34.101558, -118.340944); // Grauman's Chinese Theatre
    
    TGSearchRequest *request = [TGSearchRequest.alloc initWithSingleLine:address location:currentLocation searchExtent:nil suggestionKey:nil];
    
    [TGFindService searchWithRequest:request completionHandler:^(TGSearchResponse * _Nonnull response) {
        TGSearchResult *result = response.results.firstObject;
        if (result != nil) {
            CLLocationCoordinate2D coordinate = result.location;
            // Coordinate found!
            
            [self handleOutput:[NSValue valueWithMKCoordinate:coordinate]];
        }
    }];
}

@end

File: Swift 4/Reference App/301. Get lat long pair from street address/FindLatLongViewController.swift
//
//  FindLatLongViewController.swift
//  Reference App
//
//  Created by David Deller on 5/1/18.
//  Copyright © 2018 TallyGo. All rights reserved.
//

import UIKit
import TallyGoKit

class FindLatLongViewController: InputOutputViewController {
    
    override func handleInput(_ input: String) {
        findCoordinate(from: input)
    }

    func findCoordinate(from address: String) {
        let currentLocation = CLLocationCoordinate2D(latitude: 34.101558, longitude: -118.340944) // Grauman's Chinese Theatre
        
        let request = TGSearchRequest(singleLine: address, location: currentLocation, searchExtent: nil, suggestionKey: nil)
        
        TGFindService.search(with: request) { (response) in
            if let coordinate = response.results.first?.location {
                // Coordinate found!
                
                self.handleOutput(coordinate)
            }
        }
    }

}
